pkg/controllers/node/ipam.go

```diff
@@ -313,42 +611,21 @@ func (c *NodeController) nodeExists(knode string) bool {
 	return true
 }
 
-// podExistsOnNode returns whether the given pod exists in the Kubernetes API and is on the provided Kubernetes node.
-// Note that the "node" parameter is the name of the Kubernetes node in the Kubernetes API.
-func (c *NodeController) podExistsOnNode(name, ns, node string) bool {
-	p, err := c.k8sClientset.CoreV1().Pods(ns).Get(context.Background(), name, v1.GetOptions{})
-	if err != nil {
-		if errors.IsNotFound(err) {
-			return false
-		}
-		log.WithError(err).Warn("Failed to query pod, assume it exists")
-	}
-	if p.Spec.NodeName != node {
-		// If the pod has been rescheduled to a new node, we can treat the old allocation as
-		// gone and clean it up.
-		fields := log.Fields{"old": node, "new": p.Spec.NodeName, "pod": name, "ns": ns}
-		log.WithFields(fields).Info("Pod rescheduled on new node. Will clean up old allocation")
-		return false
-	}
-	return true
-}
-
 // kubernetesNodeForCalico returns the name of the Kubernetes node that corresponds to this Calico node.
 // This function returns an empty string if no action should be taken for this node.
```

Comment:
I think this comment is out of date now too since a return of an empty string triggers cleanup